### PR TITLE
Add heic-decode

### PR DESCRIPTION
**Pull request by hieyou1**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by hieyou1** [comment]
Very, very sorry to the people who got mentioned when I removed types/node on my sparse checkout and pushed without realizing that the change would automatically commit upstream and merge with my PR. Not what I intended to do.

File: types/heic-decode/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Type definitions for heic-decode 1.1
+// Project: https://github.com/catdad-experiments/heic-decode
+// Definitions by: hieyou1 <https://github.com/hieyou1>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ You can declare types that are available via importing the module */
```
Comment (peterblazejewicz) [inline types/heic-decode/index.d.ts:6]:
```suggestion
```
it comes from template, does it?, if so, get a rid of this instructional remark.

File: types/heic-decode/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    data: number[];
+}
+
+interface Decodable {
+    decode(): Promise<DecodedImage>;
+}
+
+/*~ If this module has methods, declare them as functions like so.
+ */
+
```
Comment (peterblazejewicz) [inline types/heic-decode/index.d.ts:23]: Same here, should not be in final code

File: types/heic-decode/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+declare function decode(bufobj: HasBuffer): Promise<DecodedImage>;
```
Comment (peterblazejewicz) [inline types/heic-decode/index.d.ts:24]:
```suggestion
/**
 * @async
 */
declare function decode(bufobj: HasBuffer): Promise<DecodedImage>;
```
declaration files does not support 'async' keyword, JSDocs serve the purpose:
https://jsdoc.app/tags-async.html

File: types/heic-decode/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+declare namespace decode {
+    function all(bufobj: HasBuffer): Promise<Decodable[]>;
```
Comment (peterblazejewicz) [inline types/heic-decode/index.d.ts:23]:
```suggestion
    /**
     * @async
     */
    function all(bufobj: HasBuffer): Promise<Decodable[]>;
```

**Comment by peterblazejewicz** [comment]
please run:
```bash
npm run prettier -- --write ./types/heic-decode/**
```
as per readme, to adhere to DT format conventions, thx!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@hieyou1 thx1

**Comment by hieyou1** [comment]
Ready to merge

**Pull request merged**